### PR TITLE
sd-device: make sd_device_new_from_path() support symlink outside of /sys

### PR DESCRIPTION
**Pull request by yuwata**
Inspired by https://github.com/systemd/systemd/pull/24265#discussion_r942261094.

### CODE REVIEW

File: src/libsystemd/sd-device/sd-device.c [hunk 1]
```diff
@@ -510,13 +510,24 @@ _public_ int sd_device_new_from_devname(sd_device **ret, const char *devname) {
...
         assert_return(path, -EINVAL);
 
         if (path_startswith(path, "/dev"))
                 return sd_device_new_from_devname(ret, path);
 
-        return sd_device_new_from_syspath(ret, path);
+        if (path_startswith(path, "/sys"))
+                return sd_device_new_from_syspath(ret, path);
+
+        /* Maybe, a symlink located outside of /sys? Let's try to chase the symlink. */
```
Comment (poettering) [inline src/libsystemd/sd-device/sd-device.c]: mayb explicitly mention the /proc/device-tree usecase here, because this might raise eyebrows otherwise

**Comment by poettering** [comment]
lgtm, but you marked it a draft

**Comment by poettering** [comment]
hmm device_set_syspath() already can do chase_symlinks(). I wonder if it would be better to rely on that and make the checking of /sys/ prefix in device_set_syspath() optional via a flag.

**Comment by yuwata** [comment]
@poettering Thank you for your review. PTAL!

**Pull request merged**